src/main/java/com/demo/classroom/Controller/AuthController.java

```diff
@@ -0,0 +1,39 @@
+package com.demo.classroom.Controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.demo.classroom.DTO.RegistrationDTO;
+import com.demo.classroom.Service.RegistrationService;
+import com.demo.classroom.Utility.ErrorMessages;
+
+import jakarta.validation.Valid;
+
+@RestController
+@RequestMapping("/req")
+public class AuthController {
+    @Autowired
+    private RegistrationService registrationService;
+
+    @PostMapping(value = "/signup", consumes = "application/json")
+    public ResponseEntity<String> registerUser(@Valid @RequestBody RegistrationDTO request, BindingResult result) {
```

Comment:
Use a name like `register` instead of `registerUser` for the method to keep it concise, especially since it's already in an `AuthController` which implies user registration.

Reply:
Consider returning a more generic response type like `ResponseEntity<?>` instead of R`esponseEntity<String> `so that you can return more complex response bodies if needed.

Reply:
done